uktrade/data-hub-frontend
# Refactored Routed Chips into a Single Component

## Description of change

Refactors the three routed chip components (`RoutedOptionFilterChips`, `RoutedDateFilterChips` and `RoutedAdviserFilterChips`) into a single `RoutedFilterChips` component.

## Test instructions

As a refactor, everything should work as before - however, I have also made the mock adviser call a bit more clever so it should now return the specific adviser that you selected instead of always showing `Puck Head`. This was needed to ensure that the route and state were consistent with each other with regards to the selected adviser.

To test that the chips are working correctly, try adding various filters, and clearing them by clicking on the chip and the clear all filters option.

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
